Extension/src/LanguageServer/configurations.ts

```diff
@@ -386,6 +395,71 @@ export class CppProperties {
         }
     }
 
+    public nodeAddonIncludesFound(): number {
+        return this.nodeAddonIncludes.length;
+    }
+
+    private async readNodeAddonIncludeLocations(rootPath: string): Promise<void> {
+        let error: Error | undefined;
+        const package_json: any = await fs.promises.readFile(path.join(rootPath, "package.json"), "utf8")
+            .then(pdj => JSON.parse(pdj))
+            .catch(e => (error = e));
+
+        const nodeAddonMap: { [dependency: string]: string } = {
+            "nan": "node --no-warnings -e \"require('nan')\"",
+            "node-addon-api": "node --no-warnings -p \"require('node-addon-api').include\""
+        };
+
+        let added: number = 0;
+        if (!error) {
+            try {
+                for (const dep in nodeAddonMap) {
+                    if (dep in package_json.dependencies) {
+                        const execCmd: string = nodeAddonMap[dep];
+                        let stdout: string = await util.execChildProcess(execCmd, rootPath);
```

Comment:
This will automatically execute any node or node.exe process in the user's root workspace if they have a package.json, which is potentially unwanted. Could you use something like
```
        const pathToNode: string = which.sync("node");
        const nodeAddonMap: { [dependency: string]: string } = {
            "nan": `"${pathToNode}" --no-warnings -e \"require('nan')\"`,
            "node-addon-api": `"${pathToNode}" --no-warnings -p \"require('node-addon-api').include\"`
        };
```
instead? That will cause the "node" on the user's path to execute instead of potentially one in their current working directory.

Reply:
no problem.